ovirt_imageio/client/_options.py

```diff
@@ -243,7 +249,9 @@ def _merge_config(self, args):
 
             # Skip options set by the command line.
             if getattr(args, option.name, None) is not None:
-                continue
+                # These store actions create default values automatically.
+                if option.action not in ['store_false', 'store_true']:
+                    continue
```

Comment:
Not very clear whats going on here. We used to skip non None option, meaning that the
user specified a value.

So now we skip these options only if option.action is not store_true or store_false.
Why?

Reply:
The problem is that even if the user did not specify a value, if the variable has a default value, it will appear as a set (not None) option. The way we solved this, is by setting the default only **after** we read the configuration file options.

However, with 'store_false' and 'store_true' actions, default bool values are set automatically when we `parse_args` to `True` and `False` respectively, before processing the configuration file, thus preventing them to be processed from the config file. I have added the condition to avoid that, and making the file configuration to prevail over the command line option.

But for these two actions should be OK. In the configuration file, it only makes sense to set `secure=` option when set to `False` (as default is `True` already), and that cannot be made `secure=True` again from the command line anyway:

Config | CLI | Result
--- | --- | ---
Missing | Missing | True
Missing | False |  False
True(1) | * | True(1)
False | * | False

(1) This option does not make much sense, as default value is already True, no need to write it in the configuration.

Reply:
Thanks, this is clear now. I think we don't want to use `action=store_xxx` since this
requires a default, and we this does not work with getting options both from the command
line and config.

We want to define the type in the option, and have `args.secure = None`. Then we can
set the default based on the config value (if set), the same way we treat other settings.